core/vm/lfvm/memory.go

```diff
@@ -31,35 +31,47 @@ func (m *Memory) ExpansionCosts(size int) uint64 {
 	return fee
 }
 
-func (m *Memory) EnsureCapacity(offset, size int, c *context) {
+func (m *Memory) EnsureCapacity(offset, size uint64, c *context) {
 	if size <= 0 {
 		return
 	}
 	needed := offset + size
-	if m.Len() < needed {
+	if uint64(m.Len()) < needed {
 		needed = toValidMemorySize(needed)
 		fee := m.ExpansionCosts(needed)
 		if !c.UseGas(fee) {
 			return
 		}
 		m.total_memory_cost += fee
-		m.store = append(m.store, make([]byte, needed-m.Len())...)
+		m.store = append(m.store, make([]byte, needed-uint64(m.Len()))...)
 	}
 }
 
-func (m *Memory) Len() int {
-	return len(m.store)
+func (m *Memory) EnsureCapacityWithoutGas(size uint64, c *context) {
+	if size <= 0 {
+		return
+	}
+	if uint64(m.Len()) < size {
+		size = toValidMemorySize(size)
+		fee := m.ExpansionCosts(size)
+		m.total_memory_cost += fee
+		m.store = append(m.store, make([]byte, size-uint64(m.Len()))...)
+	}
+}
+
+func (m *Memory) Len() uint {
```

Comment:
should `Len()` also return a uint64 since it is commonly used for size and offset values?

Reply:
done